app/1.0.x/plugin-development/admin-api.md

```diff
@@ -34,100 +34,138 @@ Kong will detect and load your endpoints if they are defined in a module named:
+  - One or more functions can be indexed with HTTP method names, like `GET` or `PUT`. These functions
+    will be executed when the appropriate HTTP method and `path` is matched. If a `before` function is
+    present on the `path`, it will be executed first. Keep in mind that `before` functions can
+    use `kong.response.exit` to finish early, effectively cancelling the "regular" http method function.
+  - Zero or one function can be indexed with `on_error`. If present, the function will be executed
+    when the code from other functions (either from a `before` or a "http method") throws an error. If
+    not present, then Kong will use a default error handler to return the errors.
+
+For example:
+
+``` lua
+local endpoints = require "kong.api.endpoints"
+
+local credentials_schema = kong.db.keyauth_credentials.schema
+local consumers_schema = kong.db.consumers.schema
 
-```lua
 return {
-  ["/my-plugin/new/get/endpoint"] = {
-    GET = function(self, dao_factory, helpers)
-      -- ...
-    end
-  }
+  ["/consumers/:consumers/key-auth"] = {
+    schema = credentials_schema,
+    methods = {
+      GET = endpoints.get_collection_endpoint(
+              credentials_schema, consumers_schema, "consumer"),
+
+      POST = endpoints.post_collection_endpoint(
+              credentials_schema, consumers_schema, "consumer"),
+    },
+  },
 }
 ```
 
-The handler function takes three arguments, which are, in order:
-
-- `self`: The request object. See [Lapis request
-  object](http://leafo.net/lapis/reference/actions.html#request-object)
-- `dao_factory`: The DAO Factory. See the
-  [datastore]({{page.book.chapters.access-the-datastore}}) chapter of this
-  guide.
-- `helpers`: A table containing a few helpers, described below.
-
-In addition to the HTTPS verbs it supports, a route table can also contain two
-other keys:
-
-- **before**: as in
-  [Lapis](http://leafo.net/lapis/reference/actions.html#handling-http-verbs), a
-  before_filter that runs before the executed verb action.
-- **on_error**: a custom error handler function that overrides the one provided
-  by Kong. See Lapis' [capturing recoverable
-  errors](http://leafo.net/lapis/reference/exception_handling.html#capturing-recoverable-errors)
-  documentation.
-
----
-
-## Helpers
-
-When handling a request on the Admin API, there are times when you want to send
-back responses and handle errors, to help you do so the third parameter
-`helpers` is a table with the following properties:
+This code will create two Admin API endpoints in `/consumers/:consumers/key-auth`, to
+obtain (`GET`) and create (`POST`) credentials associated to a given consumer. On this example
+the functions are provided by the `kong.api.endpoints` library. If you want to see a more
+complete example, with custom code in functions, see
+[the real keyauth `api.lua` file](https://github.com/Kong/kong/blob/master/kong/plugins/key-auth/api.lua).
```

Comment:
keyauth -> key-auth

(I'd write it as `the api.lua file from the key-auth plugin`)